### PR TITLE
Added SLES12 SP5 and SLES4SAP

### PR DESCRIPTION
**Pull request by deankannenberg**
## What does this PR change?

Add the needed onboarding repos entires for SLES 12 SP5 and SLES4 SAP products to spacewalk-common-channels.

- add SLES12 SP5 support for bootstrap repos
- add SLES4SAP (12 SP4, 12 SP5, 15SP2) support for bootstrap repos

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: We don't document all channels at spacewalk-common-channels.

- [x] **DONE**

## Test coverage
- No tests: Manually tested

- [x] **DONE**

## Links

None

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by juliogonzalez — CHANGES_REQUESTED** [review]
@deankannenberg I updated the template at the description for you. I guessed you tested this manually, so that's what I wrote.

The changes look good. I guess you are not adding SLES4SAP 15SP1, 12SP4 etc because the are on ESPOS, right? (https://www.suse.com/lifecycle/)

The only missing thing is the changelog entry.

You just need to edit `utils/spacewalk-utils` and add a new entry on top.

Something like, for example:

```
- Add client tools for SLES12SP5 and SLES4SAP 12SP4/12SP5/15SP2
```

**Comment by deankannenberg** [comment]
Thank you for helping with my first contribution.
Yes I only added the SLES 12 SP4 / SLES 12 SP5 / SLES 15 SP2 client tools because of the current SLES lifecycle.
I updated the "utils/spacewalk-utils.changes" file. I hope I changed it the right way.

File: utils/spacewalk-common-channels.ini [hunk 1]
```diff
@@ -1221,6 +1281,26 @@ gpgkey_id = %(_uyuni_gpgkey_id)s
 gpgkey_fingerprint = %(_uyuni_gpgkey_fingerprint)s
 repo_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/SLE15-Uyuni-Client-Tools/SLE_15/
 
+[sles15-sap-sp2-uyuni-client]
+name     = Uyuni Client Tools for SLES15 SP2 %(arch)s
+archs    =  x86_64, s390x, aarch64, ppc64le
+base_channels = sle-product-sles_sap15-sp2-pool-%(arch)s
+checksum = sha256
+gpgkey_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/SLE15-Uyuni-Client-Tools/SLE_15/repodata/repomd.xml.key
```
Comment (juliogonzalez) [inline utils/spacewalk-common-channels.ini:1289]: I guess the URL should be `https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/SLE15-Uyuni-Client-Tools/SLE_15/repodata/repomd.xml.key` here (Stable tools)...

File: utils/spacewalk-common-channels.ini [hunk 2]
```diff
@@ -1221,6 +1281,26 @@ gpgkey_id = %(_uyuni_gpgkey_id)s
...
+gpgkey_id = %(_uyuni_gpgkey_id)s
+gpgkey_fingerprint = %(_uyuni_gpgkey_fingerprint)s
+repo_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/SLE15-Uyuni-Client-Tools/SLE_15/
```
Comment (juliogonzalez) [inline utils/spacewalk-common-channels.ini:1292]: And `https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/SLE15-Uyuni-Client-Tools/SLE_15/` here.

**Review by juliogonzalez — COMMENTED** [review]
Changelog is fine :-)

I just noticed something now about SLE15 SAP SP2. Other than that, the PR LGTM.

Thanks for the contribution!

@mcalmer, can you have a second look tomorrow, just in case?

File: utils/spacewalk-common-channels.ini (hunk 1 above)
Comment (deankannenberg) [reply utils/spacewalk-common-channels.ini:1289]: I copied this url block from the other SLES15 SP2 parts above. Shall I just edit this from Master to Stable anyway?
See for e.g. line: 1249

File: utils/spacewalk-common-channels.ini (hunk 2 above)
Comment (deankannenberg) [reply utils/spacewalk-common-channels.ini:1292]: I copied this url block from the other SLES15 SP2 parts above. Shall I just edit this from Master to Stable anyway?
See for e.g. line: 1252

File: utils/spacewalk-common-channels.ini (hunk 1 above)
Comment (juliogonzalez) [reply utils/spacewalk-common-channels.ini:1289]: Seems you unearthed a bug. Yes, please, change this line to Stable, and same for 1292. I will create a PR to fix SLE15SP2 above.

Comment (deankannenberg) [reply utils/spacewalk-common-channels.ini]: changed from master to stable in dd89b5f

File: utils/spacewalk-common-channels.ini (hunk 2 above)
Comment (deankannenberg) [reply utils/spacewalk-common-channels.ini]: changed from master to stable in dd89b5f

File: utils/spacewalk-common-channels.ini (hunk 1 above)
Comment (juliogonzalez) [reply utils/spacewalk-common-channels.ini]: The PR to fix the other repositories is at https://github.com/uyuni-project/uyuni/pull/3636

**Review by juliogonzalez — APPROVED** [review]
Looks good!

Thanks again for the contribution.

**Review by mcalmer — APPROVED** [review]
Looks good so far.
Maybe something for a future PR: add this also for SLES15-SP3* ?

**Comment by deankannenberg** [comment]
I just added the SLES4SAP 15 SP3 client tools too.
See: 
f17d6a30c9763e2bfd5987d947964a8f0a81a8ec
ee7e4c6cd6280570e3bcc57d84365f627530cbe0

File: utils/spacewalk-common-channels.ini [hunk 3]
```diff
@@ -1241,6 +1321,26 @@ gpgkey_id = %(_uyuni_gpgkey_id)s
...
+gpgkey_id = %(_uyuni_gpgkey_id)s
+gpgkey_fingerprint = %(_uyuni_gpgkey_fingerprint)s
+repo_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/SLE15-Uyuni-Client-Tools/SLE_15/
+
+[sles15-sap-sp3-devel-uyuni-client]
+name     = Uyuni Client Tools for SLES15 SP3 %(arch)s (Development)
+archs    =  x86_64, s390x, aarch64, ppc64le
+base_channels = sle-product-sles_sap15-sp3-pool-%(arch)s
+checksum = sha256
+gpgkey_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/SLE15-Uyuni-Client-Tools/SLE_15/repodata/repomd.xml.key
```
Comment (juliogonzalez) [inline utils/spacewalk-common-channels.ini:1339]:
```suggestion
gpgkey_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/SLE15-Uyuni-Client-Tools/SLE_15/repodata/repomd.xml.key
```

(as this is for the devel client tools)

Comment (juliogonzalez) [inline utils/spacewalk-common-channels.ini:1339]:
```suggestion
gpgkey_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/SLE15-Uyuni-Client-Tools/SLE_15/repodata/repomd.xml.key
```

(as this is for the devel client tools)

File: utils/spacewalk-common-channels.ini [hunk 4]
```diff
@@ -1241,6 +1321,26 @@ gpgkey_id = %(_uyuni_gpgkey_id)s
...
+gpgkey_id = %(_uyuni_gpgkey_id)s
+gpgkey_fingerprint = %(_uyuni_gpgkey_fingerprint)s
+repo_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/SLE15-Uyuni-Client-Tools/SLE_15/
```
Comment (juliogonzalez) [inline utils/spacewalk-common-channels.ini:1342]:
```suggestion
repo_url = https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Master:/SLE15-Uyuni-Client-Tools/SLE_15/
```

**Review by juliogonzalez — CHANGES_REQUESTED** [review]
Two more changes, and ready to merge :-)

**Review by juliogonzalez — APPROVED** [review]
Merging as soon as tests are passing.

**Comment by juliogonzalez** [comment]
... and merged!

Thanks a lot @deankannenberg :-)

This will be part of 2021.05

**Pull request merged**